pkg/resources/cloudcontroller/util.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudcontroller
+
+import (
+	"net/url"
+
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/crd/kubermatic/v1"
+	"k8c.io/kubermatic/v2/pkg/resources"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// ExternalCloudControllerFeatureSupported checks if the
+func ExternalCloudControllerFeatureSupported(dc *kubermaticv1.Datacenter, cluster *kubermaticv1.Cluster) bool {
+	if dc.Spec.Openstack != nil {
+		// When using OpenStack external CCM with Open Telekom Cloud the creation
+		// of LBs fail as documented in the issue below:
+		// https://github.com/kubernetes/cloud-provider-openstack/issues/960
+		// Falling back to the in-tree CloudProvider mitigates the problem, even if
+		// not all features are expected to work properly (e.g.
+		// `manage-security-groups` should be set to false in cloud config, see
+		// https://kubernetes.io/docs/concepts/cluster-administration/cloud-providers/#load-balancer
+		// for more details).
+		//
+		// TODO(irozzo) This is a dirty hack to temporarily support OTC using
+		// Openstack provider, remove this when dedicated OTC support is
+		// introduced in Kubermatic.
+		return !isOTC(dc.Spec.Openstack) && OpenStackCloudControllerSupported(cluster.Spec.Version)
+	}
+
+	if dc.Spec.Hetzner != nil {
+		return cluster.Spec.Version.Minor() >= 18
+	}
```

Comment:
Should we also check here if the network is provided? Do we check that in some other place already?

Reply:
The networks will be defaulted to what is configured for the datacenter by the KKP admin. I don't see a reason to validate anything more here in this function. If anything, Seeds/datacenters should be validated to ensure a network is set.

Reply:
> If anything, Seeds/datacenters should be validated to ensure a network is set.

Can you create a follow-up issue for that?

Reply:
https://github.com/kubermatic/kubermatic/issues/6614